src/server/qgsserviceloader.h

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+                          qgsserviceloader.h
+
+  Define abstract loader class for service modules
+  -------------------
+  begin                : 2016-12-05
+  copyright            : (C) 2016 by David Marteau
+  email                : david dot marteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSERVICELOADER_H
+#define QGSSERVICELOADER_H
+
+#include <QString>
+
+class QgsServiceModule;
+class QgsServiceRegistry;
+
+/**
+ * \ingroup server
+ * QgsServiceLoader
+ * Abstract base  Class defining the native service module loader for QGIS server services
+ */
+class SERVER_EXPORT QgsServiceLoader
+{
+  public:
+
+    //! Constructor (required by SIP bindings ???)
+    QgsServiceLoader();
+
+    //! Destructor
+    virtual ~QgsServiceLoader() = 0;
```

Comment:
This class is internal  is not intended to be exposed in python.  
The constructor should not be needed but actually I found that  if a constructor was not defined then linker complains about missing symbol when using the python binding library.